crates/extensions/tedge_http_ext/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+mod actor;
+mod messages;
+
+pub use messages::*;
+
+use actor::*;
+use tedge_actors::{new_mailbox, Address, Mailbox, Recipient, RuntimeError, RuntimeHandle};
+
+pub struct HttpActorInstance {
+    actor: HttpActor,
+    mailbox: Mailbox<(usize, HttpRequest)>,
+    address: Address<(usize, HttpRequest)>,
+    clients: Vec<Recipient<Result<HttpResponse, HttpError>>>,
+}
+
+impl HttpActorInstance {
+    pub fn new(config: HttpConfig) -> Result<Self, HttpError> {
+        let actor = HttpActor::new(config)?;
+        let (mailbox, address) = new_mailbox(10);
+
+        Ok(HttpActorInstance {
+            actor,
+            mailbox,
+            address,
+            clients: vec![],
+        })
+    }
+
+    pub fn add_client(
+        &mut self,
+        client: Recipient<Result<HttpResponse, HttpError>>,
+    ) -> Recipient<HttpRequest> {
```

Comment:
Wouldn't it be a bit more natural if this function returned a tuple of `(Sender<HttpRequest>, Receiver<Result<HttpResponse, HttpError>>)`, the `Sender` for this client to send requests to the actor and the `Receiver` to get the responses on. The actor could then just cache these sender-receiver pairs for each client and keep all the `client_id` management internally.

Reply:
This might be more natural but unfortunately this cannot work, at least not with the current setting.

If the receiver is returned by the `add_client` method this implies that the caller will have a receiver per peer and will have to `select!` messages over its receivers. This is what we have currently with `mqtt_channel` and `fs_notify` creating the receivers and giving those to the mapper. Why not, but we have then to find a way to structure the consumers - to avoid what we have currently with no clear and systematic logic.

Reply:
This has been replaced by [`fn connect(&mut self, peer: &mut impl MessageBoxPort<Request, Response>)`](https://github.com/thin-edge/thin-edge.io/pull/1622/files#diff-d6192a6d562fd29bed522731372f341b715cb6560f8def001d9860f9b6de3e11R46), so the client is free to use a specific implementation for its message box receiver.